python/lsst/pipe/base/graphBuilder.py

```diff
@@ -1097,7 +1097,8 @@ def resolveDatasetRefs(
                     except MissingDatasetTypeError:
                         resolvedRefQueryResults = []
                     for resolvedRef in resolvedRefQueryResults:
-                        assert resolvedRef.dataId in refs
+                        if resolvedRef.dataId not in refs:
```

Comment:
Maybe catching `KeyError` in this case would be more efficient, avoiding searching the map twice, but it probably does not matter.

Reply:
I've seen some examples of Python dict performance on these sorts of questions being completely counterintuitive, so while I do try to avoid double-lookups (usually with `if (x := d.get(k)) is not None`, these days), that's mostly for consistency rather than any actual expectation of performance.